cmd/health-agent/probers.go

```diff
@@ -15,6 +16,7 @@ import (
 
 func setupProbers() (*proberlist.ProberList, error) {
 	pl := proberlist.New("/probers")
+	go pl.CreateAndAdd(aws.Register, "/sys/aws", 0)
```

Comment:
Calling UnregisterDirectory is no good as calling out to aws meta data could block for several minutes before returning an error leaving the empty sys/aws folder in the mean time.

What we need is a way to register a directory in two phases. The current tricorder API doesn't allow for this. I will have to change it. 

I propose that we provide a way to create a hidden DirectorySpec that doesn't show up in the metrics. After the caller has added all the desired metrics to it, they can graft the hidden DirectorySpec into the tree where it goes at which point it will be visible.

The advantage of two phase directory creation is that the user never has to see incomplete metrics.